html/dom/documents/dom-tree-accessors/named-item.html

```diff
@@ -0,0 +1,640 @@
+  },
+  {html: type => type == "img"
+    ? '<img name="foo"><form name="foo">'
+    : `<${type} name="foo">${close(type)}<img name="foo">`,
+   desc: type => `If there is ${aOrAn(type)} and another element (${type} ` +
+                 `first), a collection must be returned (name)`,
+  },
+  {html: type => type == "img"
+    ? '<form name="foo"><img name="foo">'
+    : `<img name="foo"><${type} name="foo">${close(type)}`,
+   desc: type => `If there is ${aOrAn(type)} and another element (${type} ` +
+                 `last), a collection must be returned`,
+  },
+  {html: type => `<${type} name="foo" id="bar">`,
+   desc: type => `${type} with different name and id`
+  },
+  {html: type => `<${type} name="" id="">`,
+   desc: type => `${type} with empty name/id must not be exposed`,
+  },
+  {html: type => `<${type} name=" ">`,
+   desc: type => `${type} with whitespace as name must be exposed`,
+  },
+  {html: type => `<${type} name="foo" id=" ">`,
+   desc: type => type == "img" || type == "object"
+     ? `${type} with whitespace as id must be exposed`
+     : `${type} with whitespace as id must be exposed on window but not ` +
+       `document`,
+  },
+  {html: type => `<${type} name="0">${close(type)}` +
+                 `<${type} name="42">${close(type)}` +
+                 `<${type} name="-1">${close(type)}` +
+                 `<${type} name="4294967295">${close(type)}` +
+                 `<${type} name="4294967296">${close(type)}` +
+                 `<${type} name="4294967297">${close(type)}`,
+   desc: type => `${AOrAn(type)} with numeric name`,
+  },
+  {cond: type => !unsupportedOnDocument.includes(type),
+   html: type => "",
+   fn: type => () => {
+     const doc = document.implementation.createHTMLDocument("");
+
+     const elem = doc.createElement(type);
+     elem.name = "foo";
+     doc.body.appendChild(elem);
+     // iframe is no different, because its browsing context is null
+     assert_equals(doc.foo, elem);
+   },
+   desc: type => `${type} must be exposed in detached document (name)`,
+  },
+  {html: type => type == "img"
+    ? '<div id=foo></div><img id=foo name=bar>'
+    : `<div id=foo></div><${type} id=foo>`,
+   desc: type => type == "img" || type == "object"
+     ? `If there are a div and ${type} with the same id, the ${type} must ` +
+       `be returned on document and both on window`
+     : `If there are a div and ${type} with the same id, both must be ` +
+       `returned on window and neither on document`,
+  },
+];
+
+for (const type of supportedTypes) {
+  for (const variableTest of variableTests) {
+    if (variableTest.cond && !variableTest.cond(type)) {
+      continue;
+    }
+    let obj = {};
+    obj.html = variableTest.html(type);
+    if ("fn" in variableTest) {
+      obj.fn = variableTest.fn(type);
+    }
+    obj.desc = variableTest.desc(type);
+    tests.push(obj);
+
+  }
+}
+
+for (const curTest of tests) {
+  testDiv.innerHTML = curTest.html;
```

Comment:
This doesn't work for all types of elements, e.g., `<frameset>` ends up not being inserted at all.